theano/scan_module/scan_op.py

```diff
@@ -2198,7 +2198,7 @@ def compute_all_gradients(known_grads):
                 dC_dinps_t[dx + self.n_seqs] += dC_dXtm1
         # Construct scan op
         # Seqs
-        outer_inp_seqs = [x[::-1] for x in inputs[1:1 + self.n_seqs]]
+        outer_inp_seqs = [x[:grad_steps][::-1] for x in inputs[1:1 + self.n_seqs]]
```

Comment:
```python 
i = 0
cond = True 
# ############# THE MAIN LOOP ##############
while (i < n_steps) and cond:
    # Scan loop
```
And `n_steps` can't be `0` (It is checked before in the code). So it seems that we are good with what @nouiz proposes.